stacks-signer/src/v0/signer_state.rs

```diff
@@ -981,4 +1017,73 @@ impl LocalStateMachine {
         };
         state.tx_replay_set.clone()
     }
+
+    /// Handle a possible bitcoin fork. If a fork is detetected,
+    /// return the transactions that should be replayed.
+    pub fn handle_possible_bitcoin_fork(
+        &self,
+        db: &SignerDb,
+        client: &StacksClient,
+        expected_burn_block: &NewBurnBlock,
+        prior_state_machine: &SignerStateMachine,
+        is_in_tx_replay_mode: bool,
+    ) -> Result<Option<Vec<StacksTransaction>>, SignerChainstateError> {
+        if expected_burn_block.burn_block_height <= prior_state_machine.burn_block_height
+                && expected_burn_block.consensus_hash != prior_state_machine.burn_block
+                // TODO: handle fork while still in replay
+                && !is_in_tx_replay_mode
+        {
```

Comment:
I think it's usually more clear to invert this kind of logic, so that the function exits immediately (and there's less conditional "depth" in the body of the function):

```
if expected_burn_block.burn_block_height > prior_state_machine.burn_block_height {
    // no bitcoin fork, because we're advancing the burn block height
    return Ok(None)
}
if expected_burn_block.consensus_hash == prior_state_machine.burn_block {
    // no bitcoin fork, because we're at the same burn block hash as before
    return Ok(None)
}
if is_in_tx_replay_mode {
    // TODO: handle fork while still in replay
    info!("Detected bitcoin fork while in replay mode, will not try to handle the fork")
    return Ok(None)
}
...
```